td-agent/templates/etc/init.d/td-agent.erb

```diff
@@ -0,0 +1,348 @@
+        echo "Warning: Declaring --user in \$DAEMON_ARGS has been deprecated. Use \$TD_AGENT_USER instead." 1>&2
+        shift 1
+        TD_AGENT_USER="$1"
+        ;;
+      * )
+        START_STOP_DAEMON_ARGS="${START_STOP_DAEMON_ARGS} $1"
+        ;;
+      esac
+      shift 1
+    done
+  }
+  parse_daemon_args ${DAEMON_ARGS}
+fi
+
+if [ -n "${TD_AGENT_ARGS}" ]; then
+  ORIG_TD_AGENT_ARGS="${TD_AGENT_ARGS}"
+  TD_AGENT_ARGS=""
+  parse_td_agent_args() {
+    while [ -n "$1" ]; do
+      case "$1" in
+      "--group="?* )
+        echo "Warning: Declaring --group in \$TD_AGENT_ARGS has been deprecated. Use \$TD_AGENT_GROUP instead." 1>&2
+        TD_AGENT_GROUP="${1#*=}"
+        ;;
+      "--group" )
+        echo "Warning: Declaring --group in \$TD_AGENT_ARGS has been deprecated. Use \$TD_AGENT_GROUP instead." 1>&2
+        shift 1
+        TD_AGENT_GROUP="$1"
+        ;;
+      "--user="?* )
+        echo "Warning: Declaring --user in \$TD_AGENT_ARGS has been deprecated. Use \$TD_AGENT_USER instead." 1>&2
+        TD_AGENT_USER="${1#*=}"
+        ;;
+      "--user" )
+        echo "Warning: Declaring --user in \$TD_AGENT_ARGS has been deprecated. Use \$TD_AGENT_USER instead." 1>&2
+        shift 1
+        TD_AGENT_USER="$1"
+        ;;
+      * )
+        TD_AGENT_ARGS="${TD_AGENT_ARGS} $1"
+        ;;
+      esac
+      shift 1
+    done
+  }
+  parse_td_agent_args ${ORIG_TD_AGENT_ARGS}
+fi
+
+# Arguments to run the daemon with
+TD_AGENT_ARGS="${TD_AGENT_ARGS:-${TD_AGENT_BIN_FILE} --log ${TD_AGENT_LOG_FILE} ${TD_AGENT_OPTIONS}}"
+START_STOP_DAEMON_ARGS="${START_STOP_DAEMON_ARGS}"
+
+# Exit if the package is not installed
+[ -x "${TD_AGENT_RUBY}" ] || exit 0
+
+# Source function library.
+. /etc/init.d/functions
+
+# Define LSB log_* functions.
+# Depend on lsb-base (>= 3.0-6) to ensure that this file is present.
+. /lib/lsb/init-functions
+
+# Check the user
+if [ -n "${TD_AGENT_USER}" ]; then
+  if ! getent passwd | grep -q "^${TD_AGENT_USER}:"; then
+    echo "$0: user for running ${TD_AGENT_NAME} doesn't exist: ${TD_AGENT_USER}" >&2
+    exit 1
+  fi
+  mkdir -p "$(dirname "${TD_AGENT_PID_FILE}")"
+  chown -R "${TD_AGENT_USER}" "$(dirname "${TD_AGENT_PID_FILE}")"
+  START_STOP_DAEMON_ARGS="${START_STOP_DAEMON_ARGS} --user ${TD_AGENT_USER}"
+fi
+
+if [ -n "${TD_AGENT_GROUP}" ]; then
+  if ! getent group -s files | grep -q "^${TD_AGENT_GROUP}:"; then
+    echo "$0: group for running ${TD_AGENT_NAME} doesn't exist: ${TD_AGENT_GROUP}" >&2
+    exit 1
+  fi
+  TD_AGENT_ARGS="${TD_AGENT_ARGS} --group ${TD_AGENT_GROUP}"
+fi
+
+if [ -n "${TD_AGENT_PID_FILE}" ]; then
+  mkdir -p "$(dirname "${TD_AGENT_PID_FILE}")"
+  chown -R "${TD_AGENT_USER}" "$(dirname "${TD_AGENT_PID_FILE}")"
+  TD_AGENT_ARGS="${TD_AGENT_ARGS} --daemon ${TD_AGENT_PID_FILE}"
+fi
+
+# Use jemalloc to avoid memory fragmentation
+# TODO: Use custom built version?
+if [ -f "/usr/lib64/libjemalloc.so.1" ]; then
+  export LD_PRELOAD="/usr/lib64/libjemalloc.so.1"
```

Comment:
`LD_PRELOAD` isn't needed for jemalloc since Ruby is built with `--with-jemalloc` and we don't use old version of Ruby. Even if we bundle our own jemalloc, Ruby will use it automatically by this option.

Reply:
Oh. I've confirmed that td-agent bundled ruby links jemalloc:

```log
[vagrant@localhost Packages]$ ldd /opt/td-agent/bin/ruby
	linux-vdso.so.1 =>  (0x00007ffdde7ff000)
	libruby.so.2.4 => /opt/td-agent/lib/libruby.so.2.4 (0x00007f158107e000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00000038d2800000)
	librt.so.1 => /lib64/librt.so.1 (0x00000038d2c00000)
	libjemalloc.so.1 => /usr/lib64/libjemalloc.so.1 (0x00007f1580e43000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00000038d2000000)
	libcrypt.so.1 => /lib64/libcrypt.so.1 (0x00000038d5800000)
	libm.so.6 => /lib64/libm.so.6 (0x00000038d3000000)
	libc.so.6 => /lib64/libc.so.6 (0x00000038d2400000)
	/lib64/ld-linux-x86-64.so.2 (0x000055ad0d90d000)
	libfreebl3.so => /lib64/libfreebl3.so (0x00000038d8000000)
```